# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../ocrad/gcc43.patch
# Copyright (C) 2008 - 2009 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ocrad-0.17/common.cc.vanilla	2008-05-15 16:37:23.000000000 +0200
+++ ocrad-0.17/common.cc	2008-05-15 16:37:33.000000000 +0200
@@ -16,6 +16,7 @@
 */
 
 #include <algorithm>
+#include <cstring>
 #include <cstdio>
 #include <cstdlib>
 #include "common.h"
--- ocrad-0.17/track.cc.vanilla	2008-05-15 16:38:20.000000000 +0200
+++ ocrad-0.17/track.cc	2008-05-15 16:38:46.000000000 +0200
@@ -15,6 +15,8 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <cstdio>
+
 #include <algorithm>
 #include <vector>
 
--- ocrad-0.17/rational.h.vanilla	2008-05-15 16:39:19.000000000 +0200
+++ ocrad-0.17/rational.h	2008-05-15 16:39:38.000000000 +0200
@@ -15,6 +15,8 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <string>
+
 // Rationals are kept normalized at all times.
 // Invariant = ( gcd( num, den ) == 1 && den > 0 ).
 //
--- ocrad-0.17/common.h.vanilla	2008-05-15 16:38:02.000000000 +0200
+++ ocrad-0.17/common.h	2008-05-15 16:38:10.000000000 +0200
@@ -15,6 +15,8 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <stdio.h>
+
 namespace Ocrad {
 
 extern bool verbose;
--- ocrad-0.17/page_image.cc.vanilla	2008-05-15 16:38:58.000000000 +0200
+++ ocrad-0.17/page_image.cc	2008-05-15 16:39:12.000000000 +0200
@@ -15,6 +15,8 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <limits.h>
+
 #include <algorithm>
 #include <cctype>
 #include <cstdio>
--- ocrad-0.17/character_r12.cc.vanilla	2008-05-15 16:40:02.000000000 +0200
+++ ocrad-0.17/character_r12.cc	2008-05-15 16:41:49.000000000 +0200
@@ -16,6 +16,7 @@
 */
 
 #include <cstdio>
+#include <cstdlib>
 #include <vector>
 #include "common.h"
 #include "rectangle.h"
--- ocrad-0.17/textline_r2.cc.vanilla	2008-05-15 16:42:09.000000000 +0200
+++ ocrad-0.17/textline_r2.cc	2008-05-15 16:42:19.000000000 +0200
@@ -15,6 +15,8 @@
     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
+#include <cstdlib>
+
 #include <cctype>
 #include <cstdio>
 #include <vector>
--- ocrad-0.17/textpage.cc.vanilla	2008-05-15 16:42:33.000000000 +0200
+++ ocrad-0.17/textpage.cc	2008-05-15 16:42:49.000000000 +0200
@@ -18,6 +18,7 @@
 #include <cstdio>
 #include <string>
 #include <vector>
+#include <algorithm>
 
 #include "common.h"
 #include "rectangle.h"
--- ocrad-0.17/rational.cc.vanilla	2009-04-12 20:22:58.000000000 +0200
+++ ocrad-0.17/rational.cc	2009-04-12 20:23:28.000000000 +0200
@@ -17,7 +17,7 @@
 
 #include <algorithm>
 #include <cctype>
-#include <climits>
+#include <limits.h>
 #include <cstdlib>
 #include <string>
 
--- ocrad-0.17/page_image_io.cc.vanilla	2009-04-12 20:23:40.000000000 +0200
+++ ocrad-0.17/page_image_io.cc	2009-04-12 20:23:45.000000000 +0200
@@ -17,7 +17,7 @@
 
 #include <algorithm>
 #include <cctype>
-#include <climits>
+#include <limits.h>
 #include <cstdio>
 #include <string>
 #include <vector>
